src/nvme/linux.c

```diff
@@ -122,17 +122,47 @@ int nvme_fw_download_seq(int fd, __u32 size, __u32 xfer, __u32 offset,
 	return err;
 }
 
-static int nvme_get_telemetry_log(int fd, bool create, bool ctrl, bool rae,
-				  struct nvme_telemetry_log **buf, enum nvme_telemetry_da da,
-				  size_t *size)
+enum nvme_telemetry_da nvme_get_telemetry_max_da(int fd, size_t *max_data_tx)
+{
+	struct nvme_id_ctrl id_ctrl;
+	int err = nvme_identify_ctrl(fd, &id_ctrl);
+
+	if (err) {
+		perror("identify-ctrl");
+		errno = EINVAL;
+		return 0;
```

Comment:
I think we should stay with the current pattern and just return the error status and have the a function argument for the result, as we have with the `max_data_tx` argument.

Reply:
Done as suggested!